### ISSUE TITLE
Smoking reminders to check the temperature / water / smoke

### ISSUE THREAD

**Issue by nilsbenson** [arquicanedo/barbecueboss#62]
It would be really useful for smoking if the app could wake up periodically and let you know to go check the temperature, water status (for smokers with water), and wood status to make sure everything is good. 

Another good check would be to warn the user 30 minutes or so before the timer is done to check the internal temperature of the meat so you know if the time needs to be extended.

**Comment by arquicanedo** [comment]
This is a great feature.

**Comment by nilsbenson** [comment]
I think we should consider at some point to not only have the temp check timer - but also be able to configure each probe to a set target temp so the user can be notified when the meat hits temp because based on the outside temperature it could happen well before they think it should be done.

**Comment by arquicanedo** [comment]
Yes, I agree. Probe temperature alarms are the most useful. This is the basic interface for the 2 thermometers I have.

**Comment by arquicanedo** [comment]
After writing the previous comment I was thinking that probably the most natural interface for the low-n-slow is to have unbounded timers. Everything is temperature-driven. Instead of having the DurationPicker as the main action in the low-n-slow, we could have a TemperaturePicker for the probes and simply let the time go as it needs to.

The only problem with this is if we don't reliably get temperature values from the probe (e.g., battery runs out in the thermometer). We could burn or overcook some food.

**Comment by nilsbenson** [comment]
I agree. My smoker sits outside far enough away that if I'm anywhere in the house I can't get a connection. It picks it up if I'm within about 40-50 ft., but that's not far enough for me to pick it up inside the house. The timer is more of a general thing, and makes it useful even if you don't have a thermometer. 

The way I see it
no therm:
- you can estimate the smoking time and temperature based on meat type and weight
- enter the time to smoke
- setup the smoke/water/temp check
- get notifications during the smoking to check on it so you don't forget
- when notified you can stick a thermometer in it when you think it might be ready

with thermomter:
- things work the same
- when the probe reaches the desired internal temp it notifies and you know that it's done, regardless of how much time you set on it

One of the issues I have is that the smoker sits in full sun, and in the middle of the summer it can be difficult to control the smoking temperature so the probe is a nice way to keep on top of it. If you don't have a probe, or don't have bluetooth it would still be nice to be told to check on it (or I have a tendency to forget).

**Comment by arquicanedo** [comment]
The current implementation keeps the reminder for 2 minutes after the check time has been reached. Perhaps adding vibration would be beneficial. 

Having the user clear the alarm would be a bit problematic and cumbersome.

Another option would be to create a custom screen with the alarm. The water/smoke/temp icon in the center and some vibration until you clear it.

**Comment by nilsbenson** [comment]
I was thinking we could have a view that gets pushed that has the alert information. Then the user just clicks the go button and it goes away.

This could also be used to notify users when running the background that they need to check something. 

For the low-mem devices I think it's possible too, but we will have to see.

**Comment by arquicanedo** [comment]
I'll prioritize this between tomorrow and Friday. Then cleanup the mess with the watch-specific strings.xml. This is unecessary and created a mess.

**Comment by arquicanedo** [comment]
I think the basic implementation works. The only possible improvement would be to reset the warning flags when:

- Smoke timer is reset (e.g., set for another 2 hours)
- App wakes up - the warning flags are not persisted now so there is no way of knowing whether the warning has been cleared by the user

**Comment by arquicanedo** [comment]
I think the basic implementation is ok. We just need to persist the warning flags and add the logic to decide when to warn the user after the app has been re-opened during a smoking session.
